libtransact/src/scheduler/serial/core.rs

```diff
@@ -156,40 +175,147 @@ impl SchedulerCore {
-            .expect("scheduler shared lock is poisoned");
+        if self.current_batch.is_none() {
+            let mut shared = self
+                .shared_lock
+                .lock()
+                .expect("scheduler shared lock is poisoned");
+
+            if let Some(unscheduled_batch) = shared.pop_unscheduled_batch() {
+                self.txn_queue = unscheduled_batch.batch().transactions().to_vec();
+                self.current_batch = Some(unscheduled_batch);
+            } else {
+                return Ok(());
+            }
+        }
 
-        if let Some(pair) = shared.pop_unscheduled_batch() {
-            if pair.batch().transactions().len() != 1 {
-                unimplemented!("can't handle more than one transaction per batch");
+        let transaction = self.txn_queue.pop().ok_or_else(|| {
+            CoreError::InternalError(
+                "There are no transactions left in the queue for the current batch".into(),
+            )
+        })?;
+        let transaction_id = transaction.header_signature().into();
+        let transaction_pair = match transaction.into_pair() {
+            Ok(pair) => pair,
+            Err(err) => {
+                self.invalidate_current_batch(InvalidTransactionResult {
+                    transaction_id,
+                    error_message: format!("ill-formed transaction: {}", err),
+                    error_data: vec![],
+                })?;
+                self.send_batch_result()?;
+                return Ok(());
             }
+        };
 
-            let transaction = match pair.batch().transactions()[0].clone().into_pair() {
-                Ok(transaction) => transaction,
-                Err(err) => {
-                    // An error can occur here if the header of the
-                    // transaction cannot be deserialized. If this occurs,
-                    // we panic as handling this is not currently implemented.
-                    unimplemented!("cannot handle ill-formed transaction: {}", err)
+        let context_id = match self.previous_context {
+            Some(previous_context_id) => self
+                .context_lifecycle
+                .create_context(&[previous_context_id], &self.state_id),
+            None => self.context_lifecycle.create_context(&[], &self.state_id),
+        };
+
+        self.current_txn = Some(transaction_pair.transaction().header_signature().into());
+        self.execution_tx
+            .send(ExecutionTask::new(transaction_pair, context_id))?;
+        self.next_ready = false;
+
+        Ok(())
+    }
+
+    fn invalidate_current_batch(
+        &mut self,
+        invalid_result: InvalidTransactionResult,
+    ) -> Result<(), CoreError> {
+        let current_batch_id = self
+            .current_batch
+            .as_ref()
+            .ok_or_else(|| {
+                CoreError::InternalError(
+                    "attemping to invalidate current batch but no current batch exists".into(),
```

Comment:
The surrounding errors start with a capital letter. We should be consistent one way or the other.

We should assume they will be chained together: "top-level message: slightly lower error message: even lower error message" vs. "Top-level message: Slightly lower error message: Even lower errror message". I think my preference would be on all-lower case, since these are not really sentences?

Reply:
Updated all errors to be lower case